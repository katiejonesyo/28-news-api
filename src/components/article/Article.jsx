import React from 'react';
import PropTypes from 'prop-types';

const Article = ({ url, title, author, description })  => {
    return (
        <>
            <a href={url} style={{ 
            frontSize:'8.5rem', 
            color: 'white',
            background: 'black',
            }}>
            {title}
            </a>
            {author && <p> Author: {author} </p>}
            <p>Description: {description}</p>
        </>
    ) 
};

Article.propTypes = {
    url: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    author: PropTypes.string,
    description: PropTypes.string,
  };
  export default Article;


