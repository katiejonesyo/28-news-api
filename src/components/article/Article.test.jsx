import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Article from './Article';

describe(' Article ', () => {
    afterEach(() => cleanup());
    it('renders the article', () => {
        const { asFragment } = render(<Article />);
    expect(asFragment().toMatchSnapshot());
    });
});


