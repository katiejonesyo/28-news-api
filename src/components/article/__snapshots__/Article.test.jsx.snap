// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article component renders Article 1`] = `
<DocumentFragment>
  <a
    style="color: white; background: black; text-align: center; font-family: fantasy; padding-left: 30px; padding-right: 30px;"
  />
  <p>
    Description: 
  </p>
</DocumentFragment>
`;
